### PR TITLE
DOC-1752: Document Footnotes Plugin

### PR DESCRIPTION
**Pull request by MitchCTiny**
Related Ticket: DOC-1752

Description of Changes:
* Added footnotes plugin page and relevant partials
* Added references to the page in correct places

Need to do:
- [x] Add demo page and include in footnotes.adoc
- [x] Add release note (separate PR)
- [ ] Add a note about classes and CSS (like tableofcontents)
- [ ] Provide more accurate descriptions, more detail, and improve quality of writing
- [ ] Update all cloud-related pages
- [x] Confirm premium tier/purchase plan
- [x] Add minimum version note

Pre-checks:
- [x] Branch prefixed with `feature/6/` or `hotfix/6/`
- [x] `modules/ROOT/nav.adoc` has been updated (if applicable)
- [x] Files has been included where required (if applicable)
- [x] ~Files removed have been deleted, not just excluded from the build (if applicable)~
- [x] (New product features only) Release Note added

Review:
- [ ] Documentation Team Lead has reviewed

### CODE REVIEW

**Comment by lnewson** [comment]
@MitchCTiny don't forget to update https://www.tiny.cloud/docs/tinymce/6/editor-plugin-version as well

File: modules/ROOT/pages/footnotes.adoc [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+[source,js]
+----
+tinymce.init({
+  selector: 'textarea',  // change this value according to your HTML
+  plugins: 'footnotes',
+  toolbar: 'footnotes',
+});
+----
+
+A footnotes section is automatically created and updated whenever a footnote is inserted into the {productname} editor.
```
Comment (tiny-ben-tran) [inline modules/ROOT/pages/footnotes.adoc:36]: Is the footnotes section also updated when a footnote is removed?

**Review by tiny-ben-tran — COMMENTED** [review]
- Add footnotes plugin into https://github.com/tinymce/tinymce-docs/blob/10fced24692b854104e8b26d4cf4aeb45c4f3ce4/modules/ROOT/examples/live-demos/full-featured/example.js

**Comment by MitchCTiny** [comment]
> * Add footnotes plugin into https://github.com/tinymce/tinymce-docs/blob/10fced24692b854104e8b26d4cf4aeb45c4f3ce4/modules/ROOT/examples/live-demos/full-featured/example.js

I didn't realise there was a separate file for the example, isn't it redundant?

File: modules/ROOT/pages/footnotes.adoc (hunk 1 above)
Comment (MitchCTiny) [reply modules/ROOT/pages/footnotes.adoc:36]:
```suggestion
A footnotes section is automatically created or updated whenever a footnote is inserted, reordered, or removed within the {productname} editor.
```
How does this sound?

**Comment by lnewson** [comment]
> > * Add footnotes plugin into [`10fced2`/modules/ROOT/examples/live-demos/full-featured/example.js](https://github.com/tinymce/tinymce-docs/blob/10fced24692b854104e8b26d4cf4aeb45c4f3ce4/modules/ROOT/examples/live-demos/full-featured/example.js)
> 
> I didn't realise there was a separate file for the example, isn't it redundant?

No, if you diff them they aren't the same. The example.js is used to obscure some of the demo specific setup that wouldn't make sense to customers.

**Comment by lnewson** [comment]
@MitchCTiny you've missed some pages that need updating still by the looks. I haven't had an extensive look, but I saw you added commands but didn't add them to https://www.tiny.cloud/docs/tinymce/6/editor-command-identifiers/. That could be the only page that's missing changes though given you've done toolbars and menu items 🤷

File: modules/ROOT/partials/commands/footnotes-cmds.adoc [hunk 2]
```diff
@@ -0,0 +1,13 @@
+[cols="1,3",options="header"]
+|===
+|Command |Description
+|mceInsertFootnote |Inserts a superscripted footnote number into the editor.
```
Comment (lnewson) [inline modules/ROOT/partials/commands/footnotes-cmds.adoc:4]: This description is different to what's used in the menu item/toolbar, even though they do the same thing.

File: modules/ROOT/examples/live-demos/footnotes/index.html [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    <ol>
+      <li>Place the insertion point where the new footnote should appear.<sup id="footnote_46102500311662090427490" class="mce-footnote"><a href="#footnotes_entry_46102500311662090427490">2</a></sup></li>
+      <li>Click the <em>Insert Footnote</em> icon in the toolbar, or choose <em>Insert > Insert footnote</em> from the TinyMCE menu.</li>
+      <li>A superscripted footnote number is added to the document.</li>
+      <li>And the insertion point is moved down to a new entry in the footnotes section<sup id="footnote_795387001171662097474091" class="mce-footnote"><a href="#footnotes_entry_795387001171662097474091">3</a></sup> at the bottom of the TinyMCE editor.</li>
+      <li>Add text to this new footnote entry<sup id="footnote_83105724181662097486697" class="mce-footnote"><a href="#footnotes_entry_83105724181662097486697">4</a></sup>.</li>
+    </ol>
+    <p>Each superscripted footnote number links to its respective footnote entry, and each footnote entry links back to its respective superscripted footnote number. This allows for navigation between the editor content and the footnotes section.</p>
```
Comment (MitchCTiny) [inline modules/ROOT/examples/live-demos/footnotes/index.html:12]: @brian-forte would it be a good idea to move this instructional list out of the demo and into a section on `footnotes.adoc`. This would significantly reduce the size of the demo and improve the description of footnotes on the page.

File: modules/ROOT/pages/footnotes.adoc [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+include::partial$misc/admon-premium-plugin.adoc[]
+
+include::partial$misc/admon-requires-6.2v.adoc[]
+
+The {pluginname} plugin allows the user to insert a footnote into the {productname} editor.
+
+Invoking the plugin adds a superscripted footnote number at the current insertion point, or at the end of the current selection.
+
+The plugin then places the insertion point in a new line in the footnotes section located at the bottom of the editor, allowing the user to add the footnote entry related to the newly-created footnote.
```
Comment (MitchCTiny) [inline modules/ROOT/pages/footnotes.adoc:17]: @brian-forte might we add more detail to the first line, and move the second two lines below the example?

File: modules/ROOT/examples/live-demos/footnotes/index.html (hunk 3 above)
Comment (ephox-mogran) [inline modules/ROOT/examples/live-demos/footnotes/index.html:12]: "superscripted footnote number" sounds like a superhero...

File: modules/ROOT/examples/live-demos/footnotes/index.html [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+    <div class="mce-footnotes"><hr>
+      <ol>
+        <li id="footnotes_entry_34327763141662096813220"><a class="mce-footnotes-backlink" href="#footnote_34327763141662096813220">^&nbsp;</a><span class="mce-footnotes-note">The Footnotes plugin works with TinyMCE 6.2 or later and is available with all TinyMCE Premium Plans.</span></li>
+        <li id="footnotes_entry_46102500311662090427490"><a class="mce-footnotes-backlink" href="#footnote_46102500311662090427490">^&nbsp;</a><span class="mce-footnotes-note">If there is a highlighted selection when a footnote is added, the new footnote is added to the end of the highlighted selection.&nbsp;</span></li>
+        <li id="footnotes_entry_795387001171662097474091"><a class="mce-footnotes-backlink" href="#footnote_795387001171662097474091">^&nbsp;</a><span class="mce-footnotes-note">This is a footnote entry. If the TinyMCE document is large enough to produce a scroll bar, clicking the caret prefixing this entry will scroll the document back to the line containing entry’s superscripted footnote number.</span></li>
```
Comment (ephox-mogran) [inline modules/ROOT/examples/live-demos/footnotes/index.html:21]: Should we mention that it won't move the cursor, though?

File: modules/ROOT/pages/footnotes.adoc (hunk 4 above)
Comment (ephox-mogran) [reply modules/ROOT/pages/footnotes.adoc:17]: "in a new line in the footnotes section" is a hard to parse. Should we try a different preposition. Maybe 'of' ?

File: modules/ROOT/pages/footnotes.adoc [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+:plugincode: footnotes
+:pluginminimumplan: tierthree
+
+include::partial$misc/admon-premium-plugin.adoc[]
+
+include::partial$misc/admon-requires-6.2v.adoc[]
+
+The {pluginname} plugin allows the user to insert a footnote into the {productname} editor.
+
+Invoking the plugin adds a superscripted footnote number at the current insertion point, or at the end of the current selection.
```
Comment (ephox-mogran) [inline modules/ROOT/pages/footnotes.adoc:15]: Invoking the plugin sort of makes it sound like that's all it can do. It's just one capability of the plugin. Or is this how we normally write it?

File: modules/ROOT/pages/footnotes.adoc (hunk 4 above)
Comment (ephox-mogran) [inline modules/ROOT/pages/footnotes.adoc:17]: Is `insertion point` really clearer than `cursor` here?

**Review by ephox-mogran — APPROVED** [review]

File: modules/ROOT/pages/footnotes.adoc (hunk 6 above)
Comment (brian-forte) [reply modules/ROOT/pages/footnotes.adoc:15]: Re-written in response to both comments.

File: modules/ROOT/pages/footnotes.adoc (hunk 4 above)
Comment (brian-forte) [reply modules/ROOT/pages/footnotes.adoc]: **cursor**

The UI element that shows where input will be accepted.

At a shell prompt, this has traditionally been a rectangle or underscore glyph, sometimes flashing, sometimes not.

In strings of text in GUI-centric environments (whether the string is being displayed by a text editor or word processor) the cursor is almost always a flashing vertical line.

Many shell prompts have, since the rise of GUIs, also adopted the flashing vertical line as the cursor UI element.

**insertion point**

The place within a document where input will appear.

A cursor can show where the insertion point is, but it is not the insertion point.

It is, absolutely, a fine distinction. But the insertion point is not always shown with a cursor.

**pointer** 

The object on a screen that is controlled by, and acts as the UI proxy for, the mouse (or equivalent pointing device).

Most UI systems change the pointer’s appearance on a contextual basis.

The default arrow pointer is for selecting objects and pressing on-screen buttons.

The I-beam pointer is for selecting text strings and setting the cursor in a text string.

Clicking the I-beam pointer in a text string sets a new insertion point. And the cursor is displayed at this new point to provide visual confirmation of the new insertion point.

And, in the case of the footnotes plugin, the plugin moves the insertion point to a new line in the footnotes section (and the vertical line cursor starts flashing at this new point to show this).

Comment (brian-forte) [reply modules/ROOT/pages/footnotes.adoc]: Re-written.

File: modules/ROOT/examples/live-demos/footnotes/index.html (hunk 3 above)
Comment (brian-forte) [reply modules/ROOT/examples/live-demos/footnotes/index.html:12]: To quote my commit message:

> Material from `/examples/live-demos/footnotes/index.html` edited to work as descriptive material in `footnotes.adoc`. And, so added to `footnotes.adoc`.

File: modules/ROOT/examples/live-demos/footnotes/index.html (hunk 5 above)
Comment (brian-forte) [reply modules/ROOT/examples/live-demos/footnotes/index.html:21]: Added *NOTE* admonition documenting this to `footnotes.adoc`.

File: modules/ROOT/examples/live-demos/footnotes/index.html (hunk 3 above)
Comment (brian-forte) [reply modules/ROOT/examples/live-demos/footnotes/index.html]: Also _significantly reduce[d] the size of the demo_.

File: modules/ROOT/pages/footnotes.adoc [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+include::partial$misc/admon-premium-plugin.adoc[]
+
+include::partial$misc/admon-requires-6.2v.adoc[]
+
+The {pluginname} plugin allows the user to insert a footnote into the {productname} editor.
+
+Each time a footnote is added, a superscripted footnote number is placed at the current insertion point, or at the end of the current selection.
+
+The plugin then adds a new line to the footnotes section located at the bottom of the editor and moves the insertion point to this new line. This allows the user to add the footnote entry related to the newly-created footnote.
```
Comment (MitchCTiny) [inline modules/ROOT/pages/footnotes.adoc:17]:
```suggestion
Each time a footnote is added, a superscripted footnote number is placed at the current insertion point, or at the end of the current selection.

The plugin then adds a new line to the footnotes section located at the bottom of the editor and moves the insertion point to this new line. This allows the user to instantly insert content related to the newly-created footnote in the footnote entry.
```
Should we refer to the ` footnote entry` as the location for entering content related to the respective footnote, or as the content itself within the footnotes section?

Comment (brian-forte) [reply modules/ROOT/pages/footnotes.adoc]: I’ve pushed a slightly different tweaking of the sentence you re-wrote. Like your re-write it makes the distinction between a footnote entry and its contents more apparent.

It, further, doesn’t leave the inference in place that an end-user is required to add material to a footnote entry.

I don’t have an example to hand, but I can imagine a use-case that automates the creation of footnotes, or the auto-population of footnote entry contents (or both).

File: modules/ROOT/pages/footnotes.adoc [hunk 8]
```diff
@@ -0,0 +1,110 @@
...
+
+== Known issue
+
+Footnote entries are not automatically removed after removing an image caption containing a footnote.
+
+When an image caption with an attached footnote is removed (by unchecking the _Show caption_ checkbox in the **Insert/Edit Image* dialog), the associated footnote text should be automatically removed.
+
+As of this release, however, this automatic removal does not happen.
+
+The orphan footnote text can be deleted manually. It can also be removed by clicking the `footnotesupdate` button in the toolbar.
```
Comment (MitchCTiny) [inline modules/ROOT/pages/footnotes.adoc:99]: I feel like this is more general than specifically image captions. It would be better to specify this as a use case for the `updatefootnotes` button. Seeing as this "bug" occurs from any kind of content removal (of a footnote) directly through the DOM, which is the purpose of the button. Another example of the footnotes section not automatically being updated is when you insert a footnote in a table, and then delete the row/column it exists within; the footnotes section will not automatically update in this case either. Yet another use for the updatefootnotes button.

Comment (brian-forte) [reply modules/ROOT/pages/footnotes.adoc]: To quote my commit message:

> Turned known issue re footnotes and image captions into new section on the update footnotes button.

**Comment by brian-forte** [comment]
Closing this to re-open as a docs PR.
